kratos/python_scripts/restart_utility.py

```diff
@@ -170,6 +190,61 @@ def CreateOutputFolder(self):
                 os.makedirs(folder_path)
             self.model_part.GetCommunicator().GetDataCommunicator().Barrier()
 
+    def GetRestartFiles(self):
+        """ Return a dictionary of stepID - restart_file_list dictionary that stores sets of restart files for each step."""
+        restart_files = {}
+        if os.path.isdir(self.__GetFolderPathSave()):
+            number_of_restart_files = 0
+
+            with os.scandir(path=self.__GetFolderPathSave()) as contents:
+                for entry in contents:
+                    if self.__IsRestartFile(entry):
+
+                        label = self.__ExtractFileLabel(entry.name) # Get thread ID and step ID
+                        if label[1] in restart_files:               # Check if this step has entries already
+                            restart_files[label[1]].append( entry.name )
+                        else:
+                            restart_files[label[1]] = [ entry.name ]
+
+                        number_of_restart_files += 1
+
+            # Throw a warning if the number of restart files is too large
+            if ( number_of_restart_files > 1000 ):
+                message =   "Detected " + str(len(restart_files)) + " restart files. "
+                message +=  "Consider restricting the number of restart files to keep."
+                print( message )
```

Comment:
I would leave this completely up to the user

(also I recommend using `format` for string formatting in python. Or even `f-strings`, but unfortunately those were only added in Python 3.6)

Reply:
It was my suggestion to throw a warning as leaving (forgetting) it like this can be problematic. I would suggest a warning. The problem is, the number of files that could possibly lead to issues is case-specific... Same holds for restart at each step.